frontend/pages/queries/ManageQueriesPage/ManageQueriesPage.tsx

```diff
@@ -112,34 +113,55 @@ const ManageQueriesPage = ({
   );
   const [showInheritedQueries, setShowInheritedQueries] = useState(false);
 
+  interface IQueryKeyQueriesLoadAll {
+    scope: "enhancedQueries";
+    teamId: number | undefined;
+  }
+
   const {
     data: curTeamEnhancedQueries,
     error: curTeamQueriesError,
     isFetching: isFetchingCurTeamQueries,
     refetch: refetchCurTeamQueries,
-  } = useQuery<IListQueriesResponse, Error, IEnhancedQuery[]>(
-    [{ scope: "queries", teamId: teamIdForApi }],
-    () => queriesAPI.loadAll(teamIdForApi),
+  } = useQuery<
+    IListQueriesResponse,
+    Error,
+    IEnhancedQuery[],
+    IQueryKeyQueriesLoadAll[]
+  >(
+    [{ scope: "enhancedQueries", teamId: teamIdForApi }],
+    ({ queryKey: [{ teamId }] }) =>
+      queriesAPI.loadAll(teamId).then(({ queries }) => {
+        return queries.map(enhanceQuery);
+      }),
     {
       refetchOnWindowFocus: false,
       enabled: isRouteOk,
-      select: (data) => data.queries.map(enhanceQuery),
+      staleTime: 5000,
     }
   );
 
-  // If a team is selected, fetch inherited global queries as well
+  // If a team is selected, inherit global queries
   const {
     data: globalEnhancedQueries,
     error: globalQueriesError,
     isFetching: isFetchingGlobalQueries,
     refetch: refetchGlobalQueries,
-  } = useQuery<IListQueriesResponse, Error, IEnhancedQuery[]>(
-    [{ scope: "queries", teamId: -1 }],
-    () => queriesAPI.loadAll(),
+  } = useQuery<
+    IListQueriesResponse,
+    Error,
+    IEnhancedQuery[],
+    IQueryKeyQueriesLoadAll[]
+  >(
+    [{ scope: "enhancedQueries", teamId: API_ALL_TEAMS_ID }],
+    ({ queryKey: [{ teamId }] }) =>
+      queriesAPI.loadAll(teamId).then(({ queries }) => {
+        return queries.map(enhanceQuery);
+      }),
```

Comment:
We could extract this as a named function:

```
const loadEnhancedQueries =  ({ queryKey: [{ teamId }] }: IQueryKeyQueriesLoadAll[]): Promise<IEnhancedQuery[]> =>
      queriesAPI.loadAll(teamId).then(({ queries }) => {
        return queries.map(enhanceQuery);
      });

```

Then lines 157-160 can be replaced with simply `loadEnhancedQueries,` and also in lines 132-136.

Reply:
Update - for this to work, we'll need to type that named function exactly as useQuery does internally, which will take a bit of research. For now, merging as-is, and maybe making that improvement on a later iteration.

Reply:
@jacobshandling, here's how to type the query function:

```
const loadEnhancedQueries: QueryFunction<
  IEnhancedQuery[],
  IQueryKeyQueriesLoadAll[]
> = ({ queryKey: [{ teamId }] }) =>
  queriesAPI.loadAll(teamId).then(({ queries }) => {
    return queries.map(enhanceQuery);
  });
```